lightning/src/ln/channel.rs

```diff
@@ -4082,11 +4082,28 @@ where
 		self.is_manual_broadcast = true;
 	}
 
+	/// Returns true if this channel can be resume after a restart, implying its past the initial
+	/// funding negotiation stages (and any assocated batch channels are similarly past initial
+	/// funding negotiation).
+	///
+	/// This is equivalent to saying the channel can be persisted to disk.
+	pub fn can_resume_on_restart(&self) -> bool {
+		self.channel_state.can_resume_on_reconnect()
+			&& match self.channel_state {
+				ChannelState::AwaitingChannelReady(flags) => !flags.is_waiting_for_batch(),
+				_ => true,
+			}
+	}
+
 	/// Returns true if funding_signed was sent/received and the
 	/// funding transaction has been broadcast if necessary.
-	pub fn is_funding_broadcast(&self) -> bool {
-		!self.channel_state.is_pre_funded_state()
-			&& !matches!(self.channel_state, ChannelState::AwaitingChannelReady(flags) if flags.is_set(AwaitingChannelReadyFlags::WAITING_FOR_BATCH))
+	fn is_funding_broadcast(&self) -> bool {
```

Comment:
Given the commit message, would it make sense to rename this to `is_funding_broadcastable`?